src/tests/conformance/handshake.rs

```diff
@@ -277,6 +277,110 @@ async fn reject_non_version_replies_to_version() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn reject_non_verack_replies_to_verack() {
+    // Conformance test 005.
+    //
+    // The node rejects non-Verack message as a response to initial Verack it sent.
+    //
+    // Test procedure:
+    //  For each non-verack message,
+    //
+    //  1. Expect `Version`
+    //  2. Send `Version`
+    //  3. Expect `Verack`
+    //  4. Send test message
+    //  5. Expect `Reject(Invalid)`
+    //  6. Expect connection to be terminated
+    //
+    // This test currently fails as neither Zebra nor ZCashd currently fully comply
+    // with this behaviour, so we may need to revise our expectations.
+    //
+    // ZCashd node eagerly sends messages before handshake has been concluded.
+    // Zebra node does not send Reject, but terminates the connection.
```

Comment:
This may not be implemented yet in Zebra—I had a quick peruse and couldn't find any `Reject` messages being sent.